src/dmd/cppmangle.d

```diff
@@ -1732,10 +1732,10 @@ extern(C++):
         {
             assert(t.basetype && t.basetype.ty == Tsarray);
             assert((cast(TypeSArray)t.basetype).dim);
-            version (none)
+            version (all)
             {
                 buf.writestring("Dv");
-                buf.print((cast(TypeSArray *)t.basetype).dim.toInteger()); // -- Gnu ABI v.4
+                buf.print((cast(TypeSArray)t.basetype).dim.toInteger()); // -- Gnu ABI v.4
```

Comment:
Just remove the `version` then ?

Reply:
Using `isTypeSArray()` instead of the cast will head off this sort of problem.